packages/core/src/entity/EntityStoreBase.ts

```diff
@@ -27,23 +30,150 @@ export class EntityStoreBase {
     this.addEntity(entity);
   }
 
+  public getEntryOrAsset(
```

Comment:
The bulk of my changes are here in this method. I refactored the code from `CompositionBlock` and `useComponentProps` to fetch their bound assets from a central place in the `transformBoundContent` method in core. Therefore, the deep entity fetching logic had to be moved someplace central as well.

I put it here in this method. Hopefully it covers all the rest of the cases as well.